base/set.jl

```diff
@@ -502,16 +512,27 @@ function allunique(t::Tuple)
 end
 allunique(t::Tuple{}) = true
 
+function allunique(f::F, t::Tuple) where {F}
+    length(t) < 2 && return true
+    length(t) < 32 || return _hashed_allunique(Generator(f, t))
+    return allunique(map(f, t))
+end
+
 """
     allequal(itr) -> Bool
+    allunique(f, xs)
```

Comment:
```suggestion
    allequal(f, xs)
```

Reply:
There had to be one...

One question is whether these deserve their own docstrings, or it's OK to squeeze them in with the main ones? `unique(f, xs)` gets its own, but (IMO) `allunique(f, xs)` has much less potential to surprise.